Content.Server/Disease/DiseaseZombieSystem.cs

```diff
@@ -0,0 +1,188 @@
+using Content.Server.Disease.Components;
+using Content.Server.Weapon.Melee;
+using Content.Server.Body.Components;
+using Content.Server.Body.Systems;
+using Content.Server.Popups;
+using Content.Server.Atmos.Components;
+using Content.Server.Hands.Components;
+using Content.Server.Nutrition.Components;
+using Content.Server.Mind.Components;
+using Content.Server.Chat.Managers;
+using Content.Server.Drone.Components;
+using Content.Shared.Damage;
+using Content.Shared.MobState.Components;
+using Content.Shared.Chemistry.Components;
+using Content.Shared.Hands.EntitySystems;
+using Content.Shared.Movement.EntitySystems;
+using Content.Shared.CharacterAppearance.Components;
+using Content.Shared.CharacterAppearance.Systems;
+
+namespace Content.Server.Disease
+{
+    /// <summary>
+    /// Handles zombie propagation and inherent zombie traits
+    /// </summary>
+    public sealed class DiseaseZombieSystem : EntitySystem
+    {
+        [Dependency] private readonly DiseaseSystem _disease = default!;
+        [Dependency] private readonly DamageableSystem _damageable = default!;
+        [Dependency] private readonly BloodstreamSystem _bloodstream = default!;
+        [Dependency] private readonly MovementSpeedModifierSystem _movementSpeedModifier = default!;
+        [Dependency] private readonly SharedHandsSystem _sharedHands = default!;
+        [Dependency] private readonly SharedHumanoidAppearanceSystem _sharedHumanoidAppearance = default!;
+        [Dependency] private readonly IRobustRandom _robustRandom = default!;
+        
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<DiseaseZombieComponent, ComponentInit>(OnComponentInit);
+            SubscribeLocalEvent<DiseaseZombieComponent, MeleeHitEvent>(OnMeleeHit);
+            SubscribeLocalEvent<DiseaseZombieComponent, RefreshMovementSpeedModifiersEvent>(OnRefreshMovementSpeedModifiers);
+        }
+        
+        /// <remarks>
+        /// I would imagine that if this component got assigned to something other than a mob, it would throw hella errors.
+        /// </remarks>
+        private void OnComponentInit(EntityUid uid, DiseaseZombieComponent component, ComponentInit args)
+        {
+            if (!component.ApplyZombieTraits || !HasComp<MobStateComponent>(uid))
+                return;
+
+            if (HasComp<DiseaseCarrierComponent>(uid))
+                EntityManager.RemoveComponent<DiseaseCarrierComponent>(uid);
+            if (HasComp<RespiratorComponent>(uid))
+                EntityManager.RemoveComponent<RespiratorComponent>(uid);
+            if (HasComp<BarotraumaComponent>(uid))
+                EntityManager.RemoveComponent<BarotraumaComponent>(uid);
+            if (HasComp<HungerComponent>(uid))
+                EntityManager.RemoveComponent<HungerComponent>(uid);
+            if (HasComp<ThirstComponent>(uid))
+                EntityManager.RemoveComponent<ThirstComponent>(uid);
```

Comment:
```suggestion
            EntityManager.RemoveComponent<DiseaseCarrierComponent>(uid);
            EntityManager.RemoveComponent<RespiratorComponent>(uid);
            EntityManager.RemoveComponent<BarotraumaComponent>(uid);
            EntityManager.RemoveComponent<HungerComponent>(uid);
            EntityManager.RemoveComponent<ThirstComponent>(uid);
```

this is a fairly recent engine change (so may need to update) but remcomp no longer throws if it doesn't have the component